tools/linker.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2023 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import os
+import shutil
+
+
+def create_link(source, destination, fallback_copy=False):
+    try:
+        if os.path.isdir(source):
+            os.makedirs(destination, exist_ok=True)
+            shutil.copytree(
+                source,
+                destination,
+                copy_function=_copy if fallback_copy else _link,
+                symlinks=True,
+                dirs_exist_ok=True,
+            )
+        else:
+            os.makedirs(os.path.dirname(destination), exist_ok=True)
+            (_copy if fallback_copy else _link)(source, destination)
```

Comment:
This is clever, but not very readable. Just use a simple if-else, please.

Reply:
Thanks for the compliment. Addressed in d4cbd53.